### PR TITLE
[MXNET-1385] Improved Scala Init and Macros warning messages

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Add comprehensive comments on Init and Macros stage for Scala build to get rid of Scala docs warning.
@zachgk @andrewfayres @piyushghai @aaronmarkham 

This is the phase 1 for https://github.com/apache/incubator-mxnet/issues/14081

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala [hunk 1]
```diff
@@ -63,6 +71,8 @@ private[mxnet] object CToScalaUtils {
     * optional, what is it Scala type and possibly pass in a default value
     * @param argName The name of the argument
     * @param argType Raw arguement Type description
+    * @param returnType return type of the function (Symbol/NDArray)
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala]: Consistent use of capitalization?

File: scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala [hunk 2]
```diff
@@ -31,7 +31,15 @@ private[mxnet] object CToScalaUtils {
...
     "bool" -> "Boolean")
 
-  // Convert C++ Types to Scala Types
+  /**
+    * Convert C++ Types to Scala Types
+    * @param in input raw string that contains C type docs
+    * @param argType arg type that used for error messaging
+    * @param argName arg name used for error messaging
+    * @param returnType the type that NDArray/Symbol should be
+    * @param isJava check if generating for Java
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala]: Consistent use of capitalization?

File: scala-package/macros/src/main/scala/org/apache/mxnet/GeneratorBase.scala [hunk 3]
```diff
@@ -206,16 +239,24 @@ private[mxnet] trait RandomHelpers {
     )
   }
 
-  // hacks to manage the fact that random_normal and sample_normal have
-  // non-consistent parameter naming in the back-end
-  // this first one, merge loc/scale and mu/sigma
+  /**
+    * hacks to manage the fact that random_normal and sample_normal have
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/GeneratorBase.scala]: Could `random_normal` and `sample_normal` get some kind of cross-link or should it have ticks like this?

File: scala-package/init/src/main/scala/org/apache/mxnet/init/Base.scala [hunk 4]
```diff
@@ -29,17 +29,41 @@ object Base {
...
+    * the NDArray
+    */
   type NDArrayHandle = CPtrAddress
+  /**
+    * FunctionHandle is the C pointer to
+    * the ids of the operators
+    */
   type FunctionHandle = CPtrAddress
+  /**
+    * KVStorHandle is the C pointer to
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/Base.scala]: Spelling? `KVStoreHandle`

File: scala-package/macros/src/main/scala/org/apache/mxnet/GeneratorBase.scala (hunk 3 above)
Comment (lanking520) [reply scala-package/macros/src/main/scala/org/apache/mxnet/GeneratorBase.scala]: These are internal method names used on C backend. The method used in here is essentially combine them in the front end

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 5]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
 import scala.collection.mutable.ListBuffer
 
 class LibInfo {
+  /**
+    * Get the list of the symbol ids
+    * @param symbolList pass in an empty ListBuffer and obtain a list of operator ids
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param symbolList Pass in an empty ``ListBuffer`` and obtain a list of operator IDs
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 6]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
...
+    * @return callback result
+    */
   @native def mxSymbolListAtomicSymbolCreators(symbolList: ListBuffer[SymbolHandle]): Int
+
+  /**
+    * Get the detailed information of an operator
+    * @param handle The id of the operator
+    * @param name name of the operator
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param name Name of the operator
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 7]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
...
+    * @param desc description of the operator
+    * @param numArgs number of arguments
+    * @param argNames argument names
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param argNames Argument names
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 8]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
...
+    * @param argTypes argument types
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param argTypes Argument types
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 9]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
...
+    * @param argDescs argument descriptions
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param argDescs Argument descriptions
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 10]
```diff
@@ -22,7 +22,25 @@ import org.apache.mxnet.init.Base._
...
+    * @param keyVarNumArgs Kwargs number
+    * @return callback result
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @return Callback result
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 11]
```diff
@@ -31,6 +49,18 @@ class LibInfo {
                                           argTypes: ListBuffer[String],
                                           argDescs: ListBuffer[String],
                                           keyVarNumArgs: RefString): Int
+  /**
+    * Get the name list of all operators
+    * @param names names of all operators
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @param names Names of all operators
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 12]
```diff
@@ -31,6 +49,18 @@ class LibInfo {
...
+    * @return callback result
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @return Callback result
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 13]
```diff
@@ -31,6 +49,18 @@ class LibInfo {
...
+    */
   @native def mxListAllOpNames(names: ListBuffer[String]): Int
+
+  /**
+    * get operator id from its name
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * Get operator ID from its name
```

File: scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala [hunk 14]
```diff
@@ -31,6 +49,18 @@ class LibInfo {
...
+    * @param opName Operator name
+    * @param opHandle Operator id
+    * @return callback result
```
Comment (aaronmarkham) [inline scala-package/init/src/main/scala/org/apache/mxnet/init/LibInfo.scala]:
```suggestion
    * @return Callback result
```

File: scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala [hunk 15]
```diff
@@ -141,13 +173,25 @@ private[mxnet] object TypedSymbolAPIMacro extends GeneratorBase {
 private[mxnet] object TypedSymbolRandomAPIMacro extends GeneratorBase
   with RandomHelpers {
 
-  def typeSafeAPIDefs(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Nothing] = {
+  /**
+    * Methods that check the isContrib and call code generation
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala]:
```suggestion
    * Methods that check the ``isContrib`` and call code generation
```

File: scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala [hunk 16]
```diff
@@ -77,7 +97,13 @@ private[mxnet] object SymbolMacro extends GeneratorBase {
   */
 private[mxnet] object TypedSymbolAPIMacro extends GeneratorBase {
 
-  def typeSafeAPIDefs(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Nothing] = {
+  /**
+    * Methods that check the isContrib and call code generation
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala]:
```suggestion
    * Methods that check the ``isContrib`` and call code generation
```

File: scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala [hunk 17]
```diff
@@ -41,7 +55,13 @@ private[mxnet] class AddSymbolRandomAPIs(isContrib: Boolean) extends StaticAnnot
   */
 private[mxnet] object SymbolMacro extends GeneratorBase {
 
-  def addDefs(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Nothing] = {
+  /**
+    * Methods that check the isContrib and call code generation
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/SymbolMacro.scala]:
```suggestion
    * Methods that check the ``isContrib`` and call code generation
```

File: scala-package/macros/src/main/scala/org/apache/mxnet/NDArrayMacro.scala [hunk 18]
```diff
@@ -148,8 +178,13 @@ private[mxnet] object TypedNDArrayAPIMacro extends GeneratorBase {
   */
 private[mxnet] object TypedNDArrayRandomAPIMacro extends GeneratorBase
   with RandomHelpers {
-
-  def typeSafeAPIDefs(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Nothing] = {
+  /**
+    * methods that check the isContrib and call code generation
```
Comment (aaronmarkham) [inline scala-package/macros/src/main/scala/org/apache/mxnet/NDArrayMacro.scala]:
```suggestion
    * methods that check the ``isContrib`` and call code generation
```

**Pull request merged**